src/App.tsx

```diff
@@ -1,21 +1,67 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
 
-const USER_ID = 0;
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorMessage } from './components/ErrorMessage';
+
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/FilterEnum';
+
+const USER_ID = 10336;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage('Unable to load todos');
+    }
+  }, []);
+
+  const closeError = useCallback(() => {
+    setErrorMessage('');
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    const visibleTodos = [...todos];
```

Comment:
You don't modify this array so you no need to create new variable with array spread. In this case you can use todos directly.